test/npm-event.js

```diff
@@ -94,4 +94,45 @@ const register = require('../lib/npm-event')
     t.same(job.properties.priority, 1, 'job priority')
     t.end()
   })
+
+  tap.test('accepts packages that are bigger then 1MB', async (t) => {
+    const reqPayload = JSON.stringify({payload: {
+      name: '@test/test',
+      'dist-tags': {
+        latest: '1.0.0'
+      },
+      versions: {
+        '1.0.0': {}
+      }
+    }})
+
+    server.register({
+      register,
+      options: {env, channel}
+    })
+
+    const installation = '123'
+    const secret = crypto.createHmac('sha256', env.NPMHOOKS_SECRET)
+    .update(installation)
+    .digest('hex')
+
+    const hmacPayload = crypto.createHmac('sha256', secret)
+    .update(reqPayload)
+    .digest('hex')
+
+    const {statusCode} = await server.inject({
+      method: 'POST',
+      url: `/npm/${installation}`,
+      headers: {
+        'x-npm-signature': `sha256=${hmacPayload}`,
+        'Content-Type': 'application/json',
+        'Content-Length': 20000000
```

Comment:
Correct me if I’m mistaken, but this is only testing that when we *say* that the payload is `20000000` bytes, but the request body is the JSON object in lines 99-107? — Shouldn’t we test with a real 1MB+ body?

Also, see comment above about large numbers :)

Reply:
How do I get a 1MB+ body?

Reply:
Something like this maybe? `new Buffer.alloc(1024 * 1024 * 2).toString()` for 2MB, and then stick that in an unused field on the body object?